android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -708,26 +709,32 @@ public void handleJsonEvent(final JsonReadEvent jsonReadEvent) {
                     OpenEventApp.postEventOnUIThread(new SessionDownloadEvent(true));
                     break;
                 } case ConstantStrings.SPEAKERS: {
-                    List<Speaker> speakers = objectMapper.readValue(json, new TypeReference<List<Speaker>>() {});
+                    List<Speaker> speakers = objectMapper.readValue(json, objectMapper.getTypeFactory().constructCollectionType(List.class, Speaker.class));
 
                     realmRepo.saveSpeakers(speakers).subscribe();
 
                     OpenEventApp.postEventOnUIThread(new SpeakerDownloadEvent(true));
                     break;
                 } case ConstantStrings.SPONSORS: {
-                    List<Sponsor> sponsors = objectMapper.readValue(json, new TypeReference<List<Sponsor>>() {});
+                    List<Sponsor> sponsors = objectMapper.readValue(json, objectMapper.getTypeFactory().constructCollectionType(List.class, Sponsor.class));
 
                     realmRepo.saveSponsors(sponsors).subscribe();
 
                     OpenEventApp.postEventOnUIThread(new SponsorDownloadEvent(true));
                     break;
                 } case ConstantStrings.MICROLOCATIONS: {
-                    List<Microlocation> microlocations = objectMapper.readValue(json, new TypeReference<List<Microlocation>>() {});
+                    List<Microlocation> microlocations = objectMapper.readValue(json, objectMapper.getTypeFactory().constructCollectionType(List.class, Microlocation.class));
 
                     realmRepo.saveLocations(microlocations).subscribe();
 
                     OpenEventApp.postEventOnUIThread(new MicrolocationDownloadEvent(true));
                     break;
+                }case ConstantStrings.SESSION_TYPES: {
```

Comment:
indentation is correct now just change endpoint to 'sessions/types'
Also add sample session_types file in assets folder.